simulator/mockservices/braking.py

```diff
@@ -56,14 +56,16 @@ def __init__(self, portal_callback=None):
         self.init_state()
 
     def start_rpc_service(self):
-        super().start_rpc_service()
-        self.subscribe(
-            [
-                KEY_URI_PREFIX + "/chassis.braking/1/brake_pads.front#BrakePads",
-                KEY_URI_PREFIX + "/chassis.braking/1/brake_pads.rear#BrakePads",
-            ],
-            BrakingPreconditions(self),
-        )
+        status = super().start_rpc_service()
```

Comment:
Better approach would be to write this logic within an abstract service instead of implementing these changes across all services. What we can do is, before calling the register_listener API in the abstract service, check if the mock_entity dictionary contains the service name. (Note: The mock_entity dictionary will only have the service name if it has been started successfully.) Only then we should proceed to call the register_listener API.